src/DataStructures/ModalVector.cpp

```diff
@@ -0,0 +1,115 @@
+// Distributed under the MIT License.
```

Comment:
This whole PR seems largely copy-pasta. I think we should implement most of this stuff in a macro and then use the code in both DataVector and ModalVector.

Reply:
Done